### PR TITLE
Merge gsoc

### LINKED ISSUE #1298: Merge infra branch with develop branch
## Description
As a developer
I need to merge infra and develop branch,
so that I can update the code in develop branch.

## Mocks
Not applicable

### Update [Required]
- [ ]develop branch is updated

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1 hours

### PR DESCRIPTION
**Pull request by Rimjhim28**
### Description
Merge gsoc work into develop branch 

Fixes #1298 

### Type of Change:
**Delete irrelevant options.**

- Code

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?
The whole app works as expected

### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] New and existing unit tests pass locallywith my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Review by Rimjhim28 — COMMENTED** [review]
Resolved the codacy errors as far as possible. Rest of them can either cannot be resolved or can be taken in next issues.

**Comment by debanjanah** [comment]
as per the last powerup android team meeting, the infra branch is supposed to be merged with develop, but there's still some work left in infra so the admins will look into merging infra with develop. so merging this with develop might create unwanted changes.
@mayburgos @thisum @asclepiusgal what do you think?

**Comment by Rimjhim28** [comment]
@debanjanah Can you please specify what work is left in infra. To the best of my knowledge, infra student (@geekanamika) has completed her deliverables here, same is for me too. There is one PR left to be merged but that can also be merged after merging develop and gsoc18-infra. All I need to do is change the base branch on the PR.

**Comment by debanjanah** [comment]
@Rimjhim28 I'm not certain about it that's why I thought it's better to confirm because I remember May telling that during the last meeting.

**Comment by Rimjhim28** [comment]
@debanjanah As far as I understood, May told to merge the work to develop and create new issues for pending ones ( content for Level 2, actual images for mini-games instead of placeholders) . I have mentioned you on a conversation in slack that might help you clearing the doubt.

File: PowerUp/app/src/main/java/powerup/systers/com/ui/game_activity_level2/GameLevel2Activity.java [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+            switchScreens(type);
+        }
+    }
+
+    /**
+     * Replace the current scenario with another question/answer.
+     */
+    private void updateQA() {
+        presenter.loadQuestion();
+        presenter.loadAnswer();;
```
[the remark this answers is not in the document]
Comment (debanjanah) [reply PowerUp/app/src/main/java/powerup/systers/com/ui/game_activity_level2/GameLevel2Activity.java]: @Rimjhim28 correct this
everything else is fine!

**Review by debanjanah — APPROVED** [review]

**Review by thisum — APPROVED** [review]

**Pull request merged**